### DISCUSSION TITLE
Unterstützung zur Integration des HMS16004T mit OpenDTUoB hinter Victron Multiplus II

### DISCUSSION THREAD

**Discussion by quantenquark** [hoylabs/OpenDTU-OnBattery#1389] [Ideas]
Hello everyone,
So far I have used the OpenDTUoB with an HMS16004T to support my existing PV system as a parallel system, in particular the Dynamic Power Limiter (DPL) function for phase compensation. 

The system works great and I am very happy with it.

From October to Easter, my main system is of course not enough to cover my needs. 
How could it be otherwise :-)

It would make sense to ‘marry’ both systems to maximise the yield.
My idea is to integrate the HMS behind my Victron Multiplus II (5kVA) as a microinverter.
Advantage: Support for battery charging, direct utilisation of produced power.
Disadvantage: Surpluses are fed in directly, so zero feed-in is not possible.

Now to my question: 

Is there a way to extend the OpenDTUoB control system to support this model?
In other words, a kind of ‘if-then-else’ function that queries the state of charge of the battery and dynamically sets the limits of the HMS?
Example:
- Battery < 95%: HMS power = 100%
- Battery > 95%: Throttling of the HMS via DPL based on the result from the electricity meter
- 
I already read the CAN bus of the battery for this:

![image](https://github.com/user-attachments/assets/74874126-d5c9-4f67-817c-ef8f6258229c)

I have not been able to find a suitable post in the forum so I am looking forward to your ideas and resulting options ;-)

Thank you!

Greetings 
Achim

**Comment by AndreasBoehm** [comment]
I would like to help you but i don't understand how your setup looks like or what you are trying to achieve.
Could you draw schematics that show how everything is connected with each other?

**Reply by quantenquark** [reply]
Hi Andreas, sure! 

This is my setup's today:
- Shelly 3EM
- Victron Multiplus II (single-phase)
- MPPT Solarcharger
- Cerbo GX as controller
- Pytes batteries
- OpenDTUoB with CAN-interface
- HMS1600-4T (1680Wp)


![image](https://github.com/user-attachments/assets/0f6a0bcc-f19d-4c06-a7e0-a228e798d860)

My main system is orientated east-west, while the HMS is south-facing. The Dynamic Power Limiter is an excellent addition, as it equalises a few watts or more during the course of the day :-)
It would be useful if the HMS could support the main battery during charging. However, this would require it to be connected downstream of the inverter/Multiplus II (relocation see blue arrow). 

The system recognises the downstream inverter and feeds excess energy into the battery. Any excess power is fed into the grid as it can't be blocked. (Only the high-class Fromius Inverters are communicating with Victron).

I would therefore like to see an extension to the OpenDTU that records the state of charge via the CAN bus protocol and adjusts the power of the HMS based on this and switches back to the DPL as soon as the state of charge is reached.

Admittedly, this is just theory for now, but it would already help me to have charging support up to 89 % of the battery capacity (the charging parameters change from 90 %). Simply switching off the HMS afterwards via automation is not an ideal solution. Such a feature would therefore be my favourite extension.

I am also curious to see how this idea is received in the community and whether there are other enthusiasts who operate similar systems.

73
Achim

**Reply by spcqike** [reply]
> However, this would require it to be connected downstream of the inverter/Multiplus II (relocation see blue arrow).

are you sure about that? I plan to use a multiplus setup next year. As far as I read you can connect your external inverters wherever you want. You just need to integrate your powermeter into your Victron system. It’s best to use Victron own smart meter. But you can also use a Shelly and connect it to Venus OS  

You can connect your inverters to the ESS port, as you showed. But it’s not needed. I thought. 
The advantage of connecting it to ESS port is, that you can use it while having a global power outage. But, as you wrote, you need to take care of to much power, when the battery is full.

**Reply by quantenquark** [reply]
Hi spcqike,

Thanks a lot for your message.
Well, my knowledge is based on the Victron Manuals. 

For supporting battery charging and self consuption, any 230V micro inverter is required to be connected to the Multiplus AC_OUT.

Will share some details later on.

73
Achim

**Reply by spcqike** [reply]
![image](https://github.com/user-attachments/assets/d2963ea3-204a-4961-bca1-d2bcdb926e29)

This should be a grid parallel setup. 
But without ESS. 

(Is ESS the right term? I mean without off grid charging, as the inverter will loose grid connection on power outage) 

that’s what I would like to build next year. 


https://www.photovoltaikforum.com/thread/151247-anschluss-victron-multiplus-ii-parallel-zum-netz/

**Reply by quantenquark** [reply]
Hi spcqike,

please find the source of my comment attached:
![image](https://github.com/user-attachments/assets/b9b4d42d-9c4d-41be-85b3-2faef96ee670)


As the system will fail when grid is down, the ESS can proceed operating but is limited to the specifications (see also KW vs KVA).


https://www.victronenergy.com/live/ac_coupling:start
https://communityarchive.victronenergy.com/questions/182753/frequenzverschiebung-frequenz-shifting-bei-hoymile.html

**Reply by AndreasBoehm** [reply]
I also think that a parallel setup would be the simplest and safest option. As the multiplus is made to support a setup like this it only needs to know the current powermeter readings to charge the battery via AC (from the HMS) or discharge it if needed. HMS can run at 100% all the time.

**Reply by spcqike** [reply]
Well, yes. That what I thought and what I try to explain. 

The multiplus has 3 AC connections. AC IN (grid parallel), AC Out1 (normal loads) and AC Out2(critical load, off grid ready and fully battery backuped) 

As far as I understood you can connect a grid tied inverter like hoymiles to any AC port. 

When connected to AC In (normal grid), you need an energy meter, which can be integrated into your Victron system, in your mains distribution, to get zero grid export working. This will not work on power outage. 

when connected to AC Out1, the multiplus itself can measure the currents and use excess power to charge the battery. But only excess power, that „leaves“ the multiplus, so it’s not consumed at AC Out1 nor at AC Out2. (However this can lead to battery charging while still buying energy from the grid, as everything „before“/parallel to the multiplus can’t be measured, if you have no central power meter). This also won’t work on power outage. 

When connected to AC Out2, it’s the same as AC Out1, except you are able to use this isle while there is a global power outage. In this case you must make sure to stop the inverter when the battery SoC is high. That’s what’s described in your picture. 


at least that’s what I read online. What you find in several wiring diagrams and what I would like to build:) grid parallel in the first place.
And Maybe one small AC Out2 connected inverter later on. 

Edit:
As for over charge protection while being offgrid:
The multiplus has a programmable relay, which I hope can be used to switch an AC relay that disconnects the inverter, when the SoC reaches like 80%. But, I haven’t looked into this deeper, yet. 
but I won’t use an external software solution for that. Like HA or even openDTU. I would use the multiplus, if possible, or buy or build some AC relay to monitor grid AC voltage and battery SoC and connect or disconnect the inverter accordingly.

**Reply by quantenquark** [reply]
> I also think that a parallel setup would be the simplest and safest option. As the multiplus is made to support a setup like this it only needs to know the current powermeter readings to charge the battery via AC (from the HMS) or discharge it if needed. HMS can run at 100% all the time.

Hi Andreas,

I'll give it a try and report back afterwards. But it may take a while.

73 
Achim

**Reply by greymda** [reply]
please do post some schematics as I'm also planning this sooner or later 😅
